elastictransport/elastictransport.go

```diff
@@ -351,15 +353,10 @@ func (c *Client) Perform(req *http.Request) (*http.Response, error) {
 			c.pool.OnFailure(conn)
 			c.Unlock()
 
-			// Retry on EOF errors
-			if err == io.EOF {
-				shouldRetry = true
-			}
-
 			// Retry on network errors, but not on timeout errors, unless configured
 			if err, ok := err.(net.Error); ok {
-				if (!err.Timeout() || c.enableRetryOnTimeout) && !c.disableRetry {
-					shouldRetry = true
+				if (err.Timeout() && !c.enableRetryOnTimeout) || c.disableRetry {
```

Comment:
What is the rationale behind not retrying on timeouts by default? Seems like an odd default to me. Is this the same in other clients?

From https://github.com/elastic/elastic-transport-go/issues/2#issuecomment-984476085:

> Do you see any other error that should be explicitly avoided ?

Not that I can think of. Maybe add an optional function field to Config for users to choose which errors should be retried?

Reply:
This predates me and there's the same mechanism in the python transport, I'll dig a bit deeper on that one to check the history behind it!